src/pages/Trades.js

```diff
@@ -0,0 +1,193 @@
+// React
+import { useState, useEffect } from "react";
+
+// Components
+import Navbar from "../components/Navbar.js";
+import TradeLineItem from "../components/Trades/TradeLineItem.js";
+import FilterInputModal from "../components/Trades/FilterInputModal.js";
+import TradesDropdown from "../components/Trades/TradesDropdown.js";
+import AddTradeLineItem from "../components/Trades/AddTradeLineItem.js";
+
+// Firebase
+import { onChildAdded, ref, remove } from "firebase/database";
+import { database } from "../firebase.js";
+import { auth } from "../firebase.js";
+
+// CSS
+import "./Trades.css";
+import EditTradeModal from "../components/Trades/EditTradeModal.js";
+
+const TRADES_KEY = "trades";
+
+const Trades = () => {
+  const [user, setUser] = useState("noUser");
+  const [sort, setSort] = useState("timeA");
+  const [filter, setFilter] = useState("none");
+  const [filterCat, setFilterCat] = useState("none");
+  const [showModal, setShowModal] = useState(false);
+  const [showEditModal, setShowEditModal] = useState(false);
+  const [editTrade, setEditTrade] = useState(false);
+  const [tradesArr, setTradesArr] = useState([]);
+  const [tradelines, setTradelines] = useState(<div></div>);
+  const [masterTradesArr, setMasterTradesArr] = useState([]);
+  const [isDark, setIsDark] = useState(false);
+
+  useEffect(() => {
+    const unsubscribe = auth.onAuthStateChanged(() => {
+      if (auth.currentUser === null) {
+        setUser("noUser");
+      } else {
+        setUser(auth.currentUser.uid);
+      }
+    });
+    return () => {
+      unsubscribe();
+    };
+  });
+
+  useEffect(() => {
+    let loadedTrades = [];
+    const tradesRef = ref(database, `${user}/${TRADES_KEY}`);
+
+    const unsubscribe = onChildAdded(tradesRef, (data) => {
+      loadedTrades = [...loadedTrades, { key: data.key, val: data.val() }];
+      setTradesArr(loadedTrades);
+      setMasterTradesArr(loadedTrades);
+    });
+    return () => {
+      unsubscribe();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [user]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => filterTrades(filter), [filter]);
+  useEffect(() => {
+    setTradesArr(sortTrades(tradesArr, sort));
+    setTradelines(
+      tradesArr.map((obj) => (
+        <TradeLineItem
+          key={obj.key}
+          uid={obj.key}
+          val={obj.val}
+          handleDel={handleDel}
+          handleEdit={handleEdit}
+        />
+      ))
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [tradesArr, sort]);
+
+  const sortTrades = (arr, sortMethod) => {
```

Comment:
I would include the way trades are sorted in the function name. ascending, descending, alphabetical, numerical etc.